src/index.ts

```diff
@@ -316,10 +316,34 @@ or make sure that you already have Serverless 3.x installed in your project.
 
     const funcs = this.functions;
     const promises = [];
+    const shouldUseProviderLayers =
+      (!exclude || _.isEmpty(exclude)) &&
+      (!include ||
+        _.isEmpty(include) ||
+        (include && _.isArray(include) && include.length === funcs.length));
+    // if it's not excluding functions and (not including or including all functions)
+    if (shouldUseProviderLayers) {
+      // use providers for the layers
+      const runtime = _.get(this.serverless.service, "provider.runtime");
+      const architecture = _.get(
+        this.serverless.service,
+        "provider.architecture",
+        null
+      );
+      const layerArn = this.config.layerArn
+        ? this.config.layerArn
+        : await this.getLayerArn(runtime, architecture);
+
+      if (this.serverless.service.provider["layers"]) {
+        this.serverless.service.provider["layers"].push(layerArn);
+      } else {
+        this.serverless.service.provider["layers"] = [layerArn];
+      }
```

Comment:
### * warning *:

The above code works 100% but violates your tslint rules because of accessing `["layers"]` that's the only way to do it otherwise we'd have to push a fix for `@types/serverless`

Reply:
@mrickard not without [this](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/60209) being merged and then we'd have to update the `package.json` to grab the new version

Reply:
I have suppressed the linter for those lines I will come back and fix it with a PR once that's fixed in the dependency

Reply:
@Shereef Ok, let's keep an eye on that PR. In the meantime for this one, would you be able to add a fixture to our tests that demonstrates this fix? It could be part of this PR.

Reply:
Pr merged, tests updated. linting not bypassed anymore